bpfd-operator/controllers/bpfd-agent/common.go

```diff
@@ -458,7 +458,11 @@ func ProcessMapOwnerParam(ctx context.Context,
 				return nil, fmt.Errorf("failed to get bpfd program for BpfProgram with UUID %s: %v", bpfProgramList.Items[0].GetUID(), err)
 			}
 
-			mapOwnerStatus.mapOwnerId = &prog.Id
+			kernelInfo := prog.GetKernelInfo()
+			if kernelInfo == nil {
+				return nil, fmt.Errorf("failed to process bpfd program for BpfProgram with UUID %s: %v", bpfProgramList.Items[0].GetUID(), err)
```

Comment:
NIT this and above should be UID not UUID (that's just how it's stated in the k8s spec)